src/Microsoft.DncEng.SecretManager/Readme.md

```diff
@@ -0,0 +1,196 @@
+```
+
+#### Blob Sas Uri
+```yaml
+type: azure-storage-blob-sas-uri
+parameters:
+  connectionString: SecretReference to the connection string for the account
+  container: storage container name
+  blob: blob name
+  permissions: permissions needed for the sas e.g. 'racwd'
+```
+
+#### Table Sas Uri
+```yaml
+type: azure-storage-table-sas-uri
+parameters:
+  connectionString: SecretReference to the connection string for the account
+  table: storage table name
+  permissions: permissions needed for the sas e.g. 'racwd'
+```
+
+### Event Hub Connection String
+```yaml
+type: event-hub-connection-string
+parameters:
+  subscription: azure subscription id
+  resourceGroup: azure resource group
+  namespace: event hub namespace
+  name: event hub name
+  permissions: required permissions
+```
+
+### Random Base64
+This type produces 2 values `<name>-primary` and `<name>-secondary`. The service must use both values to validate payloads, but only encode things with the primary value.
+```yaml
+type: random-base64
+parameters:
+  bytes: integer
+```
+
+### Service Bus Connection String
+```yaml
+type: service-bus-connection-string
+parameters:
+  subscription: azure subscription id
+  resourceGroup: azure resource group
+  namespace: service bus namespace
+  permissions: required permissions
+```
+
+### Sql Connection String
+```yaml
+type: sql-connection-string
+parameters:
+  adminConnection: secret reference to the sql admin connection
+  dataSource: server url
+  database: database name
+  permissions: 'admin', 'r', 'w', or 'rw'
+```
+
+### GitHub access token
+```yaml
+type: github-access-token
+parameters:
+  gitHubBotAccountSecret: secret reference to the github account this token is for
+  gitHubBotAccountName: username of the github account
+```
+
+### GitHub Application Secret
+This type produces 4 values: `<name>-app-id`, `<name>-app-private-key`, `<name>-app-secret`, and `<name>-app-hook`.
+```yaml
+type: github-app-secret
+parameters:
+  appName: name of GitHub app
+  hasPrivateKey: boolean
+  hasWebHookSecret: boolean
+  hasAppSecret: boolean
+```
+
+### GitHub Bot Account
+This type produces 3 values: `<name>-password`, `<name>-secret`, and `<name>-recovery-codes`.
+```yaml
+type: github-account
+parameters:
+  name: the account name
+```
+
+### GitHub OAuth app
+This type produces 2 values: `<name>-client-id`, and `<name>-client-secret`.
+```yaml
+type: github-oauth-secret
+parameters:
+  appName: name of application
+```
+
+### Grafana Api Key
+```yaml
+type: grafana-api-key
+parameters:
+  environment: staging or production
+```
+
+### Helix Access Token
+```yaml
+type: helix-access-token
+parameters:
+  environment: staging or production
+```
+
+### Maestro Access Token
+```yaml
+type: maestro-access-token
+parameters:
+  environment: staging or production
+```
+
+### Text
+This type should be used sparingly, and only for things that aren't actually secret.
+```yaml
+type: text
+parameters:
+  description: what the text is
+```
```

Comment:
There should be one more block for domain account:

 ### Domain Account
```yaml
type: domain-account
parameters:
  domain: domain name
  user: user account name
  description: additional description for rotation of the password
```

Reply:
Ahh yes, I will add it.